include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -27,6 +27,10 @@
 #define ONEDPL_VERSION_MINOR 7
 #define ONEDPL_VERSION_PATCH 1
 
+#if __cplusplus < 201703L
+#    error The oneDPL library requires C++17 or greater
```

Comment:
"The oneDPL library requires a C++17 compiler."

Reply:
onedpl_config is not a public header to the best of my knowledge, therefore including it and keeping the message consistently shown is under our control I think.

The condition should also include Microsoft's special macro for the language version; as far as I remember, they might not always set `__cplusplus` properly.

As for the wording: first, "the oneDPL library" means "the oneAPI Data Parallel C++ Library library" :) so I would avoid it in our documentation and messages. In this case, I would say "C++ language versions older than C++17 are no more supported by oneDPL" or "oneDPL requires the C++ language version not less than C++17" or "The earliest version of the C++ language that oneDPL supports is C++17", etc.

Reply:
I would like to use variant "oneDPL requires the C++ language version not less than C++17"
@akukanov, @MikeDvorskiy, @rarutyun are you agree?

Reply:
Also in https://github.com/oneapi-src/oneDPL/pull/642/commits/def98c66d61228b731af1691ccc9272096ed70e7 I add include of onedpl_config.h in each public header file (in some public files it was already included, in other files - no)

Reply:
> I would like to use variant "oneDPL requires the C++ language version not less than C++17" @akukanov, @MikeDvorskiy, @rarutyun are you agree?

I dont mind